### PR TITLE
Feat: custom flow support

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Part of the effort to add custom flow support #550 
In this PR I implemented the assignment API calls and added unit tests.

### CODE REVIEW

File: client/custom_flow.go [hunk 1]
```diff
@@ -101,3 +114,20 @@ func (client *ApiClient) CustomFlows(name string) ([]CustomFlow, error) {
...
+
+func (client *ApiClient) CustomFlowAssign(assignments []CustomFlowAssignment) error {
+	return client.http.Post("/custom-flow/assign", assignments, nil)
+}
+
+func (client *ApiClient) CustomFlowUnassign(assignments []CustomFlowAssignment) error {
+	return client.http.Post("/custom-flow/unassign", assignments, nil)
+}
+
+func (client *ApiClient) CustomFlowGetAssignments(assignments []CustomFlowAssignment) ([]CustomFlowAssignment, error) {
```
Comment (TomerHeber) [inline client/custom_flow.go:126]: BluePrint ID will be empty.
The response will contain the blueprint ID.

File: client/custom_flow_test.go [hunk 2]
```diff
@@ -38,11 +38,8 @@ var _ = Describe("Custom Flow Client", func() {
 				Do(func(path string, request interface{}, response *CustomFlow) {
 					*response = mockCustomFlow
 				})
-			returnedCustomFlow, _ = apiClient.CustomFlow(mockCustomFlow.Id)
-		})
-
-		It("Should send GET request", func() {
 			httpCall.Times(1)
```
Comment (TomerHeber) [inline client/custom_flow_test.go:41]: A little bit confusing from the git diff. The change is that httpCall is now part of the test.
Running in "It" has no affect on the test. (Doesn't test anything).

File: client/http/client.go [hunk 3]
```diff
@@ -54,10 +54,13 @@ func (client *HttpClient) httpResult(response *resty.Response, err error) error
 }
 
 func (client *HttpClient) Post(path string, request interface{}, response interface{}) error {
-	result, err := client.request().
-		SetBody(request).
-		SetResult(response).
-		Post(path)
+	req := client.request().SetBody(request)
```
Comment (TomerHeber) [inline client/http/client.go:57]: This is to support POST requests with nil responses. (Same implementation as the PUT requests).

File: client/custom_flow.go [hunk 4]
```diff
@@ -35,6 +35,19 @@ type CustomFlow struct {
 	IsTerragruntRunAll   bool             `json:"isTerragruntRunAll"`
 }
 
+type CustomFlowAssignmentScope string
+
+const (
+	CustomFlowOrganizationScope CustomFlowAssignmentScope = "ORGANIZATION"
```
Comment (HeverFarber) [inline client/custom_flow.go:41]: For now, you can remove the ability to configure org scope. keep the ability to add it in the future (When we will decide we support it)

**Review by HeverFarber — APPROVED** [review]

**Pull request merged**